File: gui/src/app/src/model/external/EmployeeModify.ts
import { AddressModify } from './AddressModify';
import { UserModify } from './UserModify';

export class EmployeeModify {
    id?: number;
    name: string;
    surname: string;
    dateBirth: Date;
    user: UserModify;
    address: AddressModify;
}

File: webApi/server/Domain/Models/EmployeeModify.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace server.Domain.Models
{
    [Serializable]
    public class EmployeeModify
    {
        public long Id { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Surname { get; set; }

        [Required]
        public DateTime DateBirth { get; set; }

        [Required]
        public UserModify User { get; set; }

        [Required]
        public AddressModify Address { get; set; }
    }
}
